app/connectors/SubscriptionConnector.scala

```diff
@@ -55,7 +56,10 @@ class SubscriptionConnector @Inject()(httpClient: HttpClientV2,
       .map { response =>
         response.status match {
           case CREATED => response.json.as[SubscribedResponse]
-          case CONFLICT => AlreadySubscribedResponse
+          case UNPROCESSABLE_ENTITY =>
+            response.json.as[ErrorResponse].errorDetail.errorCode match {
+              case "004" => AlreadySubscribedResponse
```

Comment:
This should be a constant inside the ErrorResponse object as it only has meaning for that class and it is like a magic value. You can name the constant: DuplicateSubmission = "004".

Reply:
Introduced a constant: `DuplicateSubmission`